test/test-signal.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+// Testcase to produce NodeReport on signal
+if (process.argv[2] === 'child') {
+  require('../').setEvents("signal");
+
+  // Exit on loss of parent process
+  process.on('disconnect', () => process.exit(2));
+
+  function myLoop() {
+    var list = [];
+    for (var i=0; i<10000000000; i++) {
+      for (var j=0; i<1000; i++) {
+        list.push(new MyRecord());
+      }
+      for (var j=0; i<1000; i++) {
+        list[j].id += 1;
+        list[j].account += 2;
+      }
+      for (var j=0; i<1000; i++) {
+        list.pop();
+      }
+    }
+  }
+
+  function MyRecord() {
+    this.name = 'foo';
+    this.id = 128;
+    this.account = 98454324;
+  }
+
+  process.send('child started');
+  myLoop();
+} else {
+  const common = require('./common.js');
+  const fork = require('child_process').fork;
+  const fs = require('fs');
+  const tap = require('tap');
+
+  if (process.platform === 'win32') {
+    tap.fail('Unsupported on Windows', { skip: true });
+    return;
+  }
+
+  const child = fork(__filename, ['child'], { 'silent': true });
+  // Wait for child to signal it is ready before sending signal
+  child.on('message', () => {
+    // Wait a bit before sending signal to allow the child to enter myLoop()
+    setTimeout(() => {
+      child.kill('SIGUSR2');
+      setTimeout(() => child.kill('SIGTERM'), 1000);
+    }, 1000);
+  });
+  child.on('exit', (code, signal) => {
+    tap.plan(4);
+    tap.equal(code, null, 'Process should not exit cleanly');
+    tap.equal(signal, 'SIGTERM', 'Process should exit with SIGTERM');
```

Comment:
FYI, windows doesn't support signals, you already saw that... so how come you get a signal status here? Well, its because libuv knows the child was killed by a (faked) SIGTERM, its actually a completely different mechanism, so when the child finally exits, with an exit status, because there are no signals on Windows, it likes, and tells us that the child exited with SIGTERM. Which isn't true, but is convenient for this kind of code.